src/Client.cpp

```diff
@@ -10,67 +10,45 @@
 #include "libursa/ZHelpers.h"
 #include "spdlog/spdlog.h"
 
-bool UrsaClient::wait_sec() {
-    // make small sleeps so the thread could
-    // terminate quickly if it's requested to do so
-    std::this_thread::sleep_for(std::chrono::milliseconds(10));
-    wait_time += 10;
-    return wait_time >= 1000;
-}
+void UrsaClient::check_task_status(const std::string &conn_id) {
+    s_send<std::string_view>(&status_socket, "status;");
+
+    std::string res_str;
+    constexpr int max_retries = 30;
+    for (int i = 0; i < max_retries && res_str.empty(); i++) {
+        try {
+            res_str = s_recv<std::string>(&status_socket);
+        } catch (const std::runtime_error &) {
+        }
+    }
 
-void UrsaClient::status_worker() {
-    if (!this->is_interactive) {
+    if (res_str.empty()) {
```

Comment:
FWIW IMO it's also ok to just block forever when ursadb doesn't respond. 30 second is already way too long for me to wait (I'm impatient like that), and 30s is a pretty arbitrary number. Also that way when you run `./ursacli -c 'do stuff'` it can wait until the server goes online.

Reply:
Also, if we can't get progress how are we supposed to execute the query :thinking:

Reply:
If I understand correctly, our query might be running but we won't be able to check the status because of lack of free workers in the case of N task concurrently running on N workers.